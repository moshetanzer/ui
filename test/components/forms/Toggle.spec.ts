import { describe, it, expect } from 'vitest'
import { UToggle } from '#components'
import type { TypeOf } from 'zod'
import ComponentRender from '../component-render'

describe('Toggle', () => {
  it.each([
    ['basic case', {}],
    ['with id', { props: { id: 'toggle id' } }],
    ['with name', { props: { name: 'Toggle name' } }],
    ['with disabled', { props: { disabled: true } }],
    ['with onIcon', { props: { onIcon: 'i-heroicons-book-open' } }],
    ['with offIcon', { props: { offIcon: 'i-heroicons-archive-box' } }],
    ['with color', { props: { color: 'red' } }],
    ['with size', { props: { size: 'sm' } }]
    // @ts-ignore
  ])('renders %s correctly', async (nameOrHtml: string, options: TypeOf<typeof UToggle.props>) => {
    const html = await ComponentRender(nameOrHtml, options, UToggle)
    expect(html).toMatchSnapshot()
  })
})
