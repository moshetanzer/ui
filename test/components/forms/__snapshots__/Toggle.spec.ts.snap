// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Toggle > renders basic case correctly 1`] = `"<button class="relative inline-flex flex-shrink-0 border-2 border-transparent disabled:cursor-not-allowed disabled:opacity-50 focus:outline-none h-5 w-9 rounded-full focus-visible:ring-2 focus-visible:ring-primary-500 dark:focus-visible:ring-primary-400 focus-visible:ring-offset-2 focus-visible:ring-offset-white dark:focus-visible:ring-offset-gray-900 bg-gray-200 dark:bg-gray-700" id="headlessui-switch-0" role="switch" type="button" tabindex="0" aria-checked="[object Object]"><span class="pointer-events-none relative inline-block rounded-full bg-white dark:bg-gray-900 shadow transform ring-0 transition ease-in-out duration-200 h-4 w-4 translate-x-0 rtl:-translate-x-0"><!--v-if--><!--v-if--></span></button>"`;

exports[`Toggle > renders with color correctly 1`] = `"<button class="relative inline-flex flex-shrink-0 border-2 border-transparent disabled:cursor-not-allowed disabled:opacity-50 focus:outline-none h-5 w-9 rounded-full focus-visible:ring-2 focus-visible:ring-red-500 dark:focus-visible:ring-red-400 focus-visible:ring-offset-2 focus-visible:ring-offset-white dark:focus-visible:ring-offset-gray-900 bg-gray-200 dark:bg-gray-700" id="headlessui-switch-5" role="switch" type="button" tabindex="0" aria-checked="[object Object]"><span class="pointer-events-none relative inline-block rounded-full bg-white dark:bg-gray-900 shadow transform ring-0 transition ease-in-out duration-200 h-4 w-4 translate-x-0 rtl:-translate-x-0"><!--v-if--><!--v-if--></span></button>"`;

exports[`Toggle > renders with disabled correctly 1`] = `"<button disabled="" class="relative inline-flex flex-shrink-0 border-2 border-transparent disabled:cursor-not-allowed disabled:opacity-50 focus:outline-none h-5 w-9 rounded-full focus-visible:ring-2 focus-visible:ring-primary-500 dark:focus-visible:ring-primary-400 focus-visible:ring-offset-2 focus-visible:ring-offset-white dark:focus-visible:ring-offset-gray-900 bg-gray-200 dark:bg-gray-700" id="headlessui-switch-2" role="switch" type="button" tabindex="0" aria-checked="[object Object]"><span class="pointer-events-none relative inline-block rounded-full bg-white dark:bg-gray-900 shadow transform ring-0 transition ease-in-out duration-200 h-4 w-4 translate-x-0 rtl:-translate-x-0"><!--v-if--><!--v-if--></span></button>"`;

exports[`Toggle > renders with id correctly 1`] = `"<button class="relative inline-flex flex-shrink-0 border-2 border-transparent disabled:cursor-not-allowed disabled:opacity-50 focus:outline-none h-5 w-9 rounded-full focus-visible:ring-2 focus-visible:ring-primary-500 dark:focus-visible:ring-primary-400 focus-visible:ring-offset-2 focus-visible:ring-offset-white dark:focus-visible:ring-offset-gray-900 bg-gray-200 dark:bg-gray-700" id="toggle id" role="switch" type="button" tabindex="0" aria-checked="[object Object]"><span class="pointer-events-none relative inline-block rounded-full bg-white dark:bg-gray-900 shadow transform ring-0 transition ease-in-out duration-200 h-4 w-4 translate-x-0 rtl:-translate-x-0"><!--v-if--><!--v-if--></span></button>"`;

exports[`Toggle > renders with name correctly 1`] = `
"<input style="position: fixed; height: 0px; padding: 0px; overflow: hidden; clip: rect(0, 0, 0, 0); white-space: nowrap; border-width: 0px; display: none;" type="checkbox" hidden="" readonly="" name="Toggle name">
<button class="relative inline-flex flex-shrink-0 border-2 border-transparent disabled:cursor-not-allowed disabled:opacity-50 focus:outline-none h-5 w-9 rounded-full focus-visible:ring-2 focus-visible:ring-primary-500 dark:focus-visible:ring-primary-400 focus-visible:ring-offset-2 focus-visible:ring-offset-white dark:focus-visible:ring-offset-gray-900 bg-gray-200 dark:bg-gray-700" id="headlessui-switch-1" role="switch" type="button" tabindex="0" aria-checked="[object Object]"><span class="pointer-events-none relative inline-block rounded-full bg-white dark:bg-gray-900 shadow transform ring-0 transition ease-in-out duration-200 h-4 w-4 translate-x-0 rtl:-translate-x-0"><!--v-if--><!--v-if--></span></button>"
`;

exports[`Toggle > renders with offIcon correctly 1`] = `"<button class="relative inline-flex flex-shrink-0 border-2 border-transparent disabled:cursor-not-allowed disabled:opacity-50 focus:outline-none h-5 w-9 rounded-full focus-visible:ring-2 focus-visible:ring-primary-500 dark:focus-visible:ring-primary-400 focus-visible:ring-offset-2 focus-visible:ring-offset-white dark:focus-visible:ring-offset-gray-900 bg-gray-200 dark:bg-gray-700" id="headlessui-switch-4" role="switch" type="button" tabindex="0" aria-checked="[object Object]"><span class="pointer-events-none relative inline-block rounded-full bg-white dark:bg-gray-900 shadow transform ring-0 transition ease-in-out duration-200 h-4 w-4 translate-x-0 rtl:-translate-x-0"><!--v-if--><span class="opacity-0 ease-out duration-100 absolute inset-0 h-full w-full flex items-center justify-center transition-opacity" aria-hidden="true"><span class="i-heroicons-archive-box h-3 w-3 text-gray-400 dark:text-gray-500"></span></span></span></button>"`;

exports[`Toggle > renders with onIcon correctly 1`] = `
"<button class="relative inline-flex flex-shrink-0 border-2 border-transparent disabled:cursor-not-allowed disabled:opacity-50 focus:outline-none h-5 w-9 rounded-full focus-visible:ring-2 focus-visible:ring-primary-500 dark:focus-visible:ring-primary-400 focus-visible:ring-offset-2 focus-visible:ring-offset-white dark:focus-visible:ring-offset-gray-900 bg-gray-200 dark:bg-gray-700" id="headlessui-switch-3" role="switch" type="button" tabindex="0" aria-checked="[object Object]"><span class="pointer-events-none relative inline-block rounded-full bg-white dark:bg-gray-900 shadow transform ring-0 transition ease-in-out duration-200 h-4 w-4 translate-x-0 rtl:-translate-x-0"><span class="opacity-100 ease-in duration-200 absolute inset-0 h-full w-full flex items-center justify-center transition-opacity" aria-hidden="true"><span class="i-heroicons-book-open h-3 w-3 text-primary-500 dark:text-primary-400"></span></span>
  <!--v-if--></span>
</button>"
`;

exports[`Toggle > renders with size correctly 1`] = `"<button class="relative inline-flex flex-shrink-0 border-2 border-transparent disabled:cursor-not-allowed disabled:opacity-50 focus:outline-none h-4 w-7 rounded-full focus-visible:ring-2 focus-visible:ring-primary-500 dark:focus-visible:ring-primary-400 focus-visible:ring-offset-2 focus-visible:ring-offset-white dark:focus-visible:ring-offset-gray-900 bg-gray-200 dark:bg-gray-700" id="headlessui-switch-6" role="switch" type="button" tabindex="0" aria-checked="[object Object]"><span class="pointer-events-none relative inline-block rounded-full bg-white dark:bg-gray-900 shadow transform ring-0 transition ease-in-out duration-200 h-3 w-3 translate-x-0 rtl:-translate-x-0"><!--v-if--><!--v-if--></span></button>"`;
